flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/split/MySqlSplitSerializer.java

```diff
@@ -161,7 +161,7 @@ public MySqlSplit deserializeSplit(int version, byte[] serialized) throws IOExce
             boolean isSuspended = false;
             if (version >= 3) {
                 totalFinishedSplitSize = in.readInt();
-                if (version > 3) {
+                if (version > 3 && in.available() > 0) {
```

Comment:
I think this code should be like :
```
if (version > 3) {
    // Descirbe the bug in PR#996.
    if (in.available() <= 0 && version != 4) {
        throw new IllegalStateException("Expected more info for the split.");
    }
    isSuspended = in.readBoolean();
}
```
IMO, change like this will not hide bugs in other future versions. But the code I provide don't look pretty. ;)